src/compiler/checker.ts

```diff
@@ -44606,15 +44613,23 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
     }
 
-    function isSymbolUsedInConditionBody(expr: Expression, body: Statement | Expression, testedNode: Node, testedSymbol: Symbol): boolean {
+    function isSymbolUsedInConditionBody(expr: Expression, body: Statement | Expression | Node, testedNode: Node, testedSymbol: Symbol, checkForAssignment: boolean): boolean {

```

Comment:
What's the idea here behind `checkForAssignment`? Why were the previous checks being done here in `isSymbolUsedInConditionBody` not enough?

Reply:
I think they would probably be fine but then

```ts
if (!a) {
  console.log(a)
}
```

wouldn't error when it probably should? But also it would be weird to reference something after checking it's falsey so maybe it would be fine and simpler just to reuse